plugins/hypervisors/vmware/src/main/java/com/cloud/hypervisor/vmware/resource/VmwareResource.java

```diff
@@ -1959,15 +1959,14 @@ protected StartAnswer execute(StartCommand cmd) {
             }
 
             // Check for hotadd settings
-            vmConfigSpec.setMemoryHotAddEnabled(vmMo.isMemoryHotAddSupported(guestOsId));
-
+            vmConfigSpec.setMemoryHotAddEnabled(vmMo.isMemoryHotAddSupported(guestOsId) && vmSpec.isEnableDynamicallyScaleVm());
```

Comment:
Thanks @DK101010, this seems to be good.
I'm also thinking, if memory hot add memory and CPU are not supported by OS and vmSpec wants the VM to be dynamically scalable, may be we should atleast log it saying this is not supported by OS. Can you please do that here.

Reply:
@harikrishna-patnala sure, shouldn't be a problem. I will add this